sycl-fusion/CMakeLists.txt

```diff
@@ -9,6 +9,9 @@ set(SYCL_JIT_BASE_DIR ${CMAKE_CURRENT_SOURCE_DIR})
 # directories, similar to how clang/CMakeLists.txt does it.
 set(LLVM_SPIRV_INCLUDE_DIRS "${LLVM_MAIN_SRC_DIR}/../llvm-spirv/include")
 
+# Set library-wide warning options.
+set(SYCL_FUSION_WARNING_FLAGS -Werror -Wall -Wextra)
```

Comment:
I'm ok with having this on by default (in particular for CI), but there should be some CMake option to turn it off. `-Werror` can be extremely annoying while debugging an issue in code or when your testing WIP changes.

Reply:
Makes sense, added!